### ISSUE TITLE
Ability to run locally?

### ISSUE THREAD

**Issue by OktayGardener** [aws/sagemaker-python-sdk#143]
Hey!

I have built a keras model using the Sagemaker API, but my development process is incredibly slow, since I have to wait 4-5 minutes after each code change in order to run my code on Sagemaker, and I would love to run this 100% locally so I can push the code that I know will work on Sagemaker.

I saw in the documentation that you should be able to set , `train_instance_type='local'`, but when I try to do this, I get the following error:

```
ClientError: An error occurred (ValidationException) when calling the CreateTrainingJob operation: 1 validation error detected: Value 'local' at 'resourceConfig.instanceType' failed to satisfy constraint: Member must satisfy enum value set: [ml.p2.xlarge, ml.m5.4xlarge, ml.m4.16xlarge, ml.p3.16xlarge, ml.m5.large, ml.p2.16xlarge, ml.c4.2xlarge, ml.c5.2xlarge, ml.c4.4xlarge, ml.c5.4xlarge, ml.c4.8xlarge, ml.c5.9xlarge, ml.c5.xlarge, ml.c4.xlarge, ml.c5.18xlarge, ml.p3.2xlarge, ml.m5.xlarge, ml.m4.10xlarge, ml.m5.12xlarge, ml.m4.xlarge, ml.m5.24xlarge, ml.m4.2xlarge, ml.p2.8xlarge, ml.m5.2xlarge, ml.p3.8xlarge, ml.m4.4xlarge]
```

and I am invoking it in the following way:

```
estimator = TensorFlow(entry_point='itemembd.py',
                               role=role,
                               training_steps=100,
                               evaluation_steps= 100,
                               train_instance_count=1,
                               train_instance_type='ml.c4.xlarge',
                               output_path='s3://ml/artifacts/itemembd',
                               
                              )

estimator.fit('s3://ml/data/itemembd', job_name='itememdb-notebook-12')
```

This feature would be **amazing** to have, since this is a huge bottleneck while I'm trying to evaluate Sagemaker for enterprise use.

**Comment by iquintero** [comment]
hi @OktayGardener 

from your conversation in the other issue (137) we figured you were running with an older sdk. I remember you posted another update but I don't see it anymore. Anyways, It looked to me like you didn't have 

docker and docker-compose installed in your system, both of which are required for local mode. Note that there is currently a bug for which I opened a PR and will merge it asap.

**Comment by OktayGardener** [comment]
Hey @iquintero, thank you so much for your reply and in the other issue, and also for the quick implementation. When pointing to S3 and building the container locally, I'm getting the following error:

```
algo-1-LGL40_1  | NotFoundError: /opt/ml/input/data/training/data.csv; No such file or directory
.........
    raise Exception("Failed to run %s, exit code: %s" % (",".join(cmd), exit_code))
Exception: Failed to run docker-compose,-f,/private/var/folders/n8/hrylchcd2vl8s0r6j2n5th9r0000gn/T/tmpTdmsPp/docker-compose.yaml,up,--build,--abort-on-container-exit, exit code: 1
```

Here's how I'm pointing to the files:

```
def train_input_fn(training_dir, params):
    return _input_fn(training_dir, 'user_item_3month_cleaned.csv')


def eval_input_fn(training_dir, params):
    return _input_fn(training_dir, 'user_item_1month_cleaned.csv')


def _input_fn(training_dir, training_filename):
    training_set = tf.contrib.learn.datasets.base.load_csv_without_header(
        filename=os.path.join(training_dir, training_filename),
        target_dtype=np.int,
        features_dtype=np.int
    )
```

It seems like there is some path issue when doing this locally. I'm not sure if it's relevant for the PR. Am I doing something wrong?

**Comment by iquintero** [comment]
hi @OktayGardener are you using the default bucket for your training? If not, you are hitting the bug that I fixed in #144  You can upgrade to the master branch, and then you can use whatever bucket you want:

```
pip install  git+https://github.com/aws/sagemaker-python-sdk
```

this fix will be released to PyPI on tuesday afternoon (PDT). so you won't have to install from the git master branch after that.

**Comment by iquintero** [comment]
This is now on PyPI (version 1.2.3). Im going to close this issue.
